src/Microsoft.AspNetCore.Routing/Matching/DfaMatcherBuilder.cs

```diff
@@ -219,7 +219,7 @@ public DfaNode BuildDfaTree()
                                 parent.Parameters = new DfaNode()
                                 {
                                     PathDepth = parent.PathDepth + 1,
-                                    Label = parent.Label + "{...}/",
+                                    Label = includeLabel ? parent.Label + "{...}/" : null,
```

Comment:
Another way to approach this would be to include a `Parent` link - then it would be possible to compute the pathdepth and label when needed.

Reply:
I considered that but there are some places where three pieces of data are used to build the label.